torch/distributed/distributed_c10d.py

```diff
@@ -191,16 +183,25 @@ def _store_based_barrier(rank, store, timeout):
     """
     store_key = "{}:{}".format(STORE_BASED_BARRIER_PREFIX, _group_count)
     store.add(store_key, 1)
+    logging.info('Added key: {} to store for rank: {}'.format(store_key, rank))
 
     # Now wait for all workers to check in with the store.
     world_size = get_world_size()
-    worker_count = int(store.get(store_key))
+    # Use 'add' instead of 'get' since for some store implementations 'add'
+    # doesn't work well with 'get'. Ideally the store implementations should
+    # be fixed, but for backward compatiblity reasons it is risky to change
+    # the store implementations. Once, we completely migrate away from these
+    # legacy stores, we can use 'get' here instead.
+    worker_count = store.add(store_key, 0)
     start = time.time()
     while worker_count != world_size:
         time.sleep(0.01)
-        worker_count = int(store.get(store_key))
+        worker_count = store.add(store_key, 0)
```

Comment:
Also just noticed, previously this line had `int` cast but now with `add` it does not, is this a bug or do `get` and `add` return different types (str vs int) and is that intentional design of the store api (cc @osalpekar)?

Reply:
This is actually the expected behavior - the default type for the value when `set`/`get` are used is `const std::vector<uint8_t>&` so we must cast to int. In the `add` case, we're simply bumping a counter so the underlying value is stored as an `int64_t` (which is what add returns).